### PR TITLE
Add French, German, and Spanish sentence beginnings assessment

### LINKED ISSUE #746: Sentence beginning assessment for German, French and Spanish
We should add a sentence beginning assessment for German, French and Spanish.

### PR DESCRIPTION
**Pull request by IreneStr**
Adapts sentence beginnings assessment for French, German and Spanish. 
Fixes #746 

For testing: 
- Change the locale to fr_FR, de_DE or es_ES. 
- Write three consecutive sentences beginning with a first word exception from that language, each time followed by a different word. These sentences should not be marked/result in a red bullet.
- Write three consecutive sentences beginning with a first word exception from that language, each time followed by the same word. These sentences should be marked/result in a red bullet.
- Write three consecutive sentences all beginning with the same word that is not included in the exception list. These sentences should be marked/result in a red bullet as well.

### CODE REVIEW

File: js/assessments/sentenceBeginningsAssessment.js [hunk 1]
```diff
@@ -68,7 +69,7 @@ var sentenceBeginningMarker = function( paper, researcher ) {
 	sentenceBeginnings = filter( sentenceBeginnings, function( sentenceBeginning ) {
 		return sentenceBeginning.count > maximumConsecutiveDuplicates;
 	} );
-	var sentences = map ( sentenceBeginnings, function( begin ) {
+	var sentences = map( sentenceBeginnings, function( begin ) {
```
Comment (atimmer) [inline js/assessments/sentenceBeginningsAssessment.js]: This line is not covered by the specs.

File: js/researches/spanish/firstWordExceptions.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+/**
+ * Returns an array with exceptions for the sentence beginning researcher.
+ * @returns {Array} The array filled with exceptions.
+ */
+module.exports = function() {
+	return [ "el", "los", "la", "las", "un", "una", "unos", "unas", "unos", "dos", "tres", "cuatro", "cinco", "seis",
```
Comment (atimmer) [inline js/researches/spanish/firstWordExceptions.js]: `unos` is duplicated.

File: js/researches/french/firstWordExceptions.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+/**
+ * Returns an array with exceptions for the sentence beginning researcher.
+ * @returns {Array} The array filled with exceptions.
+ */
+module.exports = function() {
+	return [ "le", "la", "les", "un", "une", "deux", "trois", "quatre", "cinq", "six", "sept", "huit", "neuf",
```
Comment (atimmer) [inline js/researches/french/firstWordExceptions.js]: I would categorize these words and add a comment before all groups description what the words are so it is clear why they are exceptions.

File: js/researches/german/firstWordExceptions.js [hunk 4]
```diff
@@ -0,0 +1,11 @@
+/**
+ * Returns an array with exceptions for the sentence beginning researcher.
+ * @returns {Array} The array filled with exceptions.
+ */
+module.exports = function() {
+	return [ "acht", "das", "dem", "den", "denen", "der", "deren", "derer", "des", "dessen", "die", "diese", "diesem",
```
Comment (atimmer) [inline js/researches/german/firstWordExceptions.js]: I would categorize these words and add a comment before all groups description what the words are so it is clear why they are exceptions.

File: js/researches/spanish/firstWordExceptions.js (hunk 2 above)
Comment (atimmer) [inline js/researches/spanish/firstWordExceptions.js]: I would categorize these words and add a comment before all groups description what the words are so it is clear why they are exceptions.

File: spec/researches/getSentenceBeginningsSpec.js [hunk 5]
```diff
@@ -2,66 +2,125 @@ var sentenceBeginnings = require( "../../js/researches/getSentenceBeginnings.js"
...
 	} );
-	it( "returns an object with sentence beginnings and counts for two sentences starting with the same word.", function() {
-		var mockPaper = new Paper( "Hey, hey! Hey." );
+	it( "returns an object with sentence beginnings and counts for two sentences in English starting with the same word.", function() {
+		var mockPaper = new Paper( "Hey, hey! Hey.", { locale: 'en_US'} );
 		expect( sentenceBeginnings( mockPaper )[0].word ).toBe( "hey" );
 		expect( sentenceBeginnings( mockPaper )[0].count ).toBe( 2 );
 	} );
-	it( "returns an object with sentence beginnings and counts for four sentences, the first two starting with the same word. The fourth is starting with the same word as the first two. " +
+	it( "returns an object with sentence beginnings and counts for four sentencesin English , the first two starting with the same word. The fourth is starting with the same word as the first two. " +
```
Comment (atimmer) [inline spec/researches/getSentenceBeginningsSpec.js]: There is a missing space between `sentences` and `in`

File: spec/researches/getSentenceBeginningsSpec.js [hunk 6]
```diff
@@ -2,66 +2,125 @@ var sentenceBeginnings = require( "../../js/researches/getSentenceBeginnings.js"
...
+		expect( sentenceBeginnings( mockPaper )[0].count ).toBe( 3 );
+	} );
+	it( "returns an object with sentence beginnings and counts for two sentences in French starting with different words.", function() {
+		var mockPaper = new Paper( "Sur le pont d'Avignon. Liberté, égalité, fraternité. ", { locale: 'fr_FR'} );
+		expect( sentenceBeginnings( mockPaper )[0].word ).toBe( "sur" );
+		expect( sentenceBeginnings( mockPaper )[0].count ).toBe( 1 );
+		expect( sentenceBeginnings( mockPaper )[1].word ).toBe( "liberté" );
+		expect( sentenceBeginnings( mockPaper )[1].count ).toBe( 1 );
+	} );
+	it( "returns an object with sentence beginnings and counts for two sentences in French starting with the same word.", function() {
```
Comment (atimmer) [inline spec/researches/getSentenceBeginningsSpec.js]: I would love some whitespace between the end of the previous spec and the start of the next.

**Comment by atimmer** [comment]
CR Done 👷🏻

File: js/researches/english/firstWordExceptions.js [hunk 7]
```diff
@@ -3,5 +3,14 @@
  * @returns {Array} The array filled with exceptions.
  */
 module.exports = function() {
-	return [ "a", "an", "the", "this", "that", "these", "those", "one", "two", "three", "four", "five", "six", "seven", "eight", "nine", "ten" ];
+	return [
+		// Definite articles:
+		"the",
+		// Indefinite articles:
+		"a", "an",
+		// Numbers 1-10 ('une' is already included in the indefinite articles):
```
Comment (atimmer) [inline js/researches/english/firstWordExceptions.js]: I don't see any `une` in this list.

**Comment by atimmer** [comment]
CR Done 👍🏻

**Comment by terw-dan** [comment]
Acceptance :+1:

**Pull request merged**